pallet-chainlink-feed/src/lib.rs

```diff
@@ -0,0 +1,1357 @@
+		/// The oracle was already enabled.
+		AlreadyEnabled,
+		/// The oracle address cannot change its associated admin.
+		OwnerCannotChangeAdmin,
+		/// Only the owner of a feed can change the configuration.
+		NotFeedOwner,
+		/// Only the pending owner of a feed can accept the transfer invitation.
+		NotPendingOwner,
+		/// The specified min/max pair was invalid.
+		WrongBounds,
+		/// The maximum number of oracles cannot exceed the amount of available oracles.
+		MaxExceededTotal,
+		/// The round initiation delay cannot be equal to or greater
+		/// than the number of oracles.
+		DelayNotBelowCount,
+		/// Sender is not admin. Admin privilege can only be transfered by the admin.
+		NotAdmin,
+		/// Only the pending admin can accept the transfer.
+		NotPendingAdmin,
+		/// The requester cannot request a new round, yet.
+		CannotRequestRoundYet,
+		/// No requester permissions associated with the given account.
+		NotAuthorizedRequester,
+		/// Cannot withdraw funds.
+		InsufficientFunds,
+		/// Funds cannot be withdrawn as the reserve would be critically low.
+		InsufficientReserve,
+		/// Only the pallet admin account can call this extrinsic.
+		NotPalletAdmin,
+		/// Only the pending admin can accept the transfer.
+		NotPendingPalletAdmin,
+		/// Round zero is not allowed to be pruned.
+		CannotPruneRoundZero,
+		/// The given pruning bounds don't cause any pruning with the current state.
+		NothingToPrune,
+		/// There is no valid data, yet, so the conditions for pruning are not met.
+		NoValidRoundYet,
+		/// The pruning should leave the rounds story in a contiguous state (no gaps).
+		PruneContiguously,
+		/// The maximum number of feeds was reached.
+		FeedLimitReached,
+		/// The round cannot be superseded by a new round.
+		NotSupersedable,
+		/// The round cannot be started because it is not a valid new round.
+		InvalidRound,
+		/// The calling account is not allowed to create feeds.
+		NotFeedCreator,
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn deposit_event() = default;
+
+		/// The account used to pay oracles and manage the funds of this pallet.
+		const FundAccount: T::AccountId = T::ModuleId::get().into_account();
+
+		// TODO: weights
+
+		// --- feed operations ---
+
+		/// Create a new oracle feed with the given config values.
+		/// Limited to feed creator accounts.
+		#[weight = T::WeightInfo::create_feed(oracles.len() as u32)]
+		pub fn create_feed(
+			origin,
+			payment: BalanceOf<T>,
+			timeout: T::BlockNumber,
+			submission_value_bounds: (T::Value, T::Value),
+			min_submissions: u32,
+			decimals: u8,
+			description: Vec<u8>,
+			restart_delay: RoundId,
+			oracles: Vec<(T::AccountId, T::AccountId)>,
+		) -> DispatchResultWithPostInfo {
+			let owner = ensure_signed(origin)?;
+			ensure!(FeedCreators::<T>::contains_key(&owner), Error::<T>::NotFeedCreator);
+			ensure!(description.len() as u32 <= T::StringLimit::get(), Error::<T>::DescriptionTooLong);
```

Comment:
Can `description.len() as u32` overflow?

Reply:
no, `usize` is `u32` in wasm